### ISSUE TITLE
Bad global gitconfig breaks git_repository

### ISSUE THREAD

**Issue by keith** [bazelbuild/bazel#9974]
Currently if a user has a `~/.gitconfig` file with invalid config, `git_repository` rules fail. I would expect this file to instead be ignored since it isn't explicitly part of the build.

### Bugs: what's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

Add these lines to your `~/.gitconfig` file:

```
[url "foobar"]
  insteadOf = https://github.com
```

Then run `bazel fetch //...` in a WORKSPACE that uses `git_repository` with github URLs.

### What operating system are you running Bazel on?

macOS

### What's the output of `bazel info release`?

release 0.29.1

**Comment by aehlig** [comment]
> Currently if a user has a `~/.gitconfig` file with invalid config, `git_repository` rules fail. I would expect this file to instead be ignored since it isn't explicitly part of the build.

On the other hand, [users expect a global `gitconfig` to be honored](https://github.com/bazelbuild/bazel/issues/3893#issue-264842109), e.g., to take proxy settings from there. These seem incompatible requirements to me.

**Comment by keith** [comment]
Yea when I submitted this I was worried there were some requirements there. I do wonder if that info should be provided to bazel in some other way, like an explicit gitconfig that is passed, although maybe that won't work for all cases.

**Comment by kastiglione** [comment]
Yeah, it seems like either Bazel provides some sort of affordance for specifying git auth or git network config relevant to setting up workspaces. Or, Bazel has to have some sort of policy around known and hermeticity holes. Given the number of hermeticity holes we see, maybe these should all be documented somewhere (independent of this issue).

**Comment by philwo** [comment]
I think the current behavior is ok, but agree we might want to document this somewhere.

**Comment by keith** [comment]
for others who hit this, you can workaround this by having `--repo_env=GIT_CONFIG_GLOBAL=/dev/null` in your `.bazelrc` file if you know you shouldn't ever need folks to override this. I don't think this is safe to do in bazel but every company I've worked at has had at least 1 person with a bad global gitconfig caused by weird recommendations by golang to do the `insteadOf` stuff
